src/services/FileService.ts

```diff
@@ -0,0 +1,271 @@
+    const { services } = await this.getServices();
+    
+    const file = await this.getFileById(fileId, userId);
+    if (!file) {
+      return false;
+    }
+
+    try {
+      // Delete from S3
+      await this.getMinioService().deleteFile(this.bucketName, file.objectKey);
+      
+      // Delete from database
+      await services.em.removeAndFlush(file);
+      return true;
+    } catch (error) {
+      console.error('Error deleting file:', error);
+      return false;
+    }
+  }
+
+  /**
+   * List files by directory with pagination
+   */
+  async listFiles(
+    userId: number,
+    path?: string,
+    page: number = 1,
+    limit: number = 20
+  ): Promise<Page<File>> {
+    const { services } = await this.getServices();
+    
+    const offset = (page - 1) * limit;
+
+    // Build where clause
+    const whereClause: any = { userId };
+    if (path) {
+      whereClause.path = path;
+    }
+
+    const findAndCount = await services.file.findAndCount(whereClause, {
+      orderBy: { createdAt: 'DESC' },
+      limit,
+      offset,
+    });
+
+    return toPageDTO(findAndCount, page, limit);
+  }
+
+  /**
+   * Get all files in a directory tree (recursive)
+   */
+  async getDirectoryTree(userId: number, path: string = '/'): Promise<File[]> {
+    const { services } = await this.getServices();
+    
+    // Get all files that start with the given path
+    const files = await services.file.find({
+      userId,
+      path: { $like: `${path}%` }
+    }, {
+      orderBy: { path: 'ASC', name: 'ASC' }
+    });
+
+    return files;
+  }
+
+  /**
+   * Create a directory (virtual - just a file with no size)
+   */
+  async createDirectory(name: string, path: string, userId: number): Promise<File> {
+    const { services } = await this.getServices();
+    
+    this.validatePath(path);
+    
+    // Ensure path ends with /
+    const normalizedPath = path.endsWith('/') ? path : `${path}/`;
+    
+    const directory = new File();
+    directory.name = name;
+    directory.objectKey = `directory_${Date.now()}_${Math.random().toString(36).substr(2, 9)}`;
+    directory.path = normalizedPath;
+    directory.size = 0;
+    directory.mimeType = 'application/x-directory';
+    directory.userId = userId;
+    
+    await services.em.persistAndFlush(directory);
+    return directory;
+  }
+
+  /**
+   * Move file to different directory
+   */
+  async moveFile(fileId: number, userId: number, newPath: string): Promise<File | null> {
+    const { services } = await this.getServices();
+    
+    const file = await this.getFileById(fileId, userId);
+    if (!file) {
+      return null;
+    }
+
+    this.validatePath(newPath);
+    
+    // Ensure new path ends with / for directories
+    if (file.mimeType === 'application/x-directory') {
+      file.path = newPath.endsWith('/') ? newPath : `${newPath}/`;
+    } else {
+      file.path = newPath;
+    }
+
+    await services.em.flush();
+    return file;
+  }
+
+  /**
```

Comment:
search we just search by name for simple juss select * from files where name like %q%